Sources/_InternalTestSupport/TestingHelpers.swift

```diff
@@ -0,0 +1,69 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2014-2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Basics
+import TSCTestSupport
+import Testing
+
+public func expectThrowsCommandExecutionError<T>(
+    _ expression: @autoclosure () async throws -> T,
+    sourceLocation: SourceLocation = #_sourceLocation,
+    _ errorHandler: (_ error: CommandExecutionError) -> Void = { _ in }
+) async {
+    let error = await #expect(throws: SwiftPMError.self, sourceLocation: sourceLocation) {
+        try await expression()
+    }
+
+    guard case .executionFailure(let processError, let stdout, let stderr) = error,
+          case AsyncProcessResult.Error.nonZeroExit(let processResult) = processError,
+          processResult.exitStatus != .terminated(code: 0) else {
+        Issue.record("Unexpected error type: \(error?.interpolationDescription)", sourceLocation: sourceLocation)
+        return
+    }
+    errorHandler(CommandExecutionError(result: processResult, stdout: stdout, stderr: stderr))
+}
+
+private func expectMatchImpl<Pattern, Value>(_ result: Bool, _ value: Value, _ pattern: Pattern, negativeMatch: Bool = false, sourceLocation: SourceLocation) {
```

Comment:
suggestion: can we omit the `expectMatch*` functions in favour of using the standard `#expect` calls?

Reply:
I think there's value in having the full set of regex based matching options available in a Swift Testing context, even if we're not using the full feature set here